packages/fluentui/react-northstar/src/components/Toolbar/ToolbarMenu.tsx

```diff
@@ -73,103 +73,129 @@ export interface ToolbarMenuProps extends UIComponentProps, ChildrenComponentPro
+    const mergedVariables = mergeComponentVariables(parentVariables, variables);
+
+    const getA11yProps = useAccessibility(accessibility, {
+      debugName: composeOptions.displayName,
+      actionHandlers: {
+        performClick: e => {
+          _.invoke(props, 'onClick', e, props);
+        },
+      },
+      rtl: context.rtl,
+    });
+    const { classes } = useStyles<ToolbarMenuStylesProps>(composeOptions.displayName, {
+      className: composeOptions.className,
+      composeOptions,
+      mapPropsToInlineStyles: () => ({
+        className,
+        design,
+        styles,
+        variables: mergedVariables,
+      }),
+      rtl: context.rtl,
+      unstable_props: props,
+    });
 
-  const parentVariables = React.useContext(ToolbarVariablesContext);
-  const mergedVariables = mergeComponentVariables(parentVariables, variables);
+    const handleItemOverrides = predefinedProps => ({
+      onClick: (e, itemProps) => {
+        _.invoke(predefinedProps, 'onClick', e, itemProps);
+        _.invoke(props, 'onItemClick', e, {
+          ...itemProps,
+          menuOpen: !!itemProps.menu,
+        });
+      },
+    });
 
-  const getA11yProps = useAccessibility(accessibility, {
-    debugName: ToolbarMenu.displayName,
-    actionHandlers: {
-      performClick: e => {
-        _.invoke(props, 'onClick', e, props);
+    const handleRadioGroupOverrides = (predefinedProps: ToolbarMenuRadioGroupProps) => ({
+      onItemClick: (e, itemProps) => {
+        _.invoke(predefinedProps, 'onItemClick', e, itemProps);
+        _.invoke(props, 'onItemClick', e, itemProps);
       },
-    },
-    rtl: context.rtl,
-  });
-  const { classes } = useStyles<ToolbarMenuStylesProps>(ToolbarMenu.displayName, {
-    className: toolbarMenuClassName,
-    mapPropsToInlineStyles: () => ({
-      className,
-      design,
-      styles,
-      variables: mergedVariables,
-    }),
-    rtl: context.rtl,
-  });
-
-  const handleItemOverrides = predefinedProps => ({
-    onClick: (e, itemProps) => {
-      _.invoke(predefinedProps, 'onClick', e, itemProps);
-      _.invoke(props, 'onItemClick', e, {
-        ...itemProps,
-        menuOpen: !!itemProps.menu,
+    });
+
+    const renderItems = () => {
+      return _.map(items, item => {
+        const kind = _.get(item, 'kind', 'item');
+
+        switch (kind) {
+          case 'divider':
+            return createShorthand(ToolbarMenuDivider, item);
+
+          case 'group':
+            return createShorthand(ToolbarMenuRadioGroup, item, { overrideProps: handleRadioGroupOverrides });
+
+          case 'toggle':
+            return createShorthand(ToolbarMenuItem, item, {
+              defaultProps: () => ({ accessibility: toolbarMenuItemCheckboxBehavior }),
+              overrideProps: handleItemOverrides,
+            });
+
+          default:
+            return createShorthand(ToolbarMenuItem, item, {
+              defaultProps: () => ({
```

Comment:
And please add everywhere `slotProps` and move whatever possible to the `mapPropsToSlotProps` config